### DISCUSSION TITLE
Is rbatis capable of retriev/insert nested structs?

### DISCUSSION THREAD

**Discussion by frederikhors** [rbatis/rbatis#330] [Q&A]
I'm trying to understand if `rbatis` is useful for what I need (I'm coming from Golang where I used https://github.com/uptrace/bun).

Is it possible to retrieve (and maybe insert) nested structs?

Example:

```rust
struct Player {
    id: String,
    created_at: time::OffsetDateTime,
    description: String,
    coach_id: String,
    coach: Coach, // or coach: Option<Box<DomainCoach>>
    teams: Option<Vec<Team>>,
    countries: Vec<Country>
}
```

I tried SeaORM but [they are stucked with only one nested (they call it related) struct](https://www.sea-ql.org/SeaORM/docs/basic-crud/select/#find-related-models) because of https://www.sea-ql.org/blog/2022-05-14-faq-01/.

I'm wondering if this is the same with rbatis.

__Thank you very much for your work!__

**Comment by zhuxiujia** [comment]
> I'm trying to understand if `rbatis` is useful for what I need (I'm coming from Golang where I used https://github.com/uptrace/bun).
> 
> Is it possible to retrieve (and maybe insert) nested structs?
> 
> Example:
> 
> ```rust
> struct Player {
>     id: String,
>     created_at: time::OffsetDateTime,
>     description: String,
>     coach_id: String,
>     coach: Coach, // or coach: Option<Box<DomainCoach>>
>     teams: Option<Vec<Team>>,
>     countries: Vec<Country>
> }
> ```
> 
> I tried SeaORM but [they are stucked with only one nested (they call it related) struct](https://www.sea-ql.org/SeaORM/docs/basic-crud/select/#find-related-models) because of https://www.sea-ql.org/blog/2022-05-14-faq-01/.
> 
> I'm wondering if this is the same with rbatis.
> 
> **Thank you very much for your work!**

yes，see [table_extend](https://github.com/rbatis/rbatis/blob/master/example/src/table_extend.rs)
